### PR TITLE
feat(CloudFront): Service and Checks

### PR DESCRIPTION
**Pull request by jfagoagas**
### Context 

Prowler 2 --> Prowler 3 Migration

### Description

- CloudFront Service
- Migrate the following checks:

Prowler v3 | Prowler v2
-- | --
cloudfront_distributions_field_level_encryption_enabled | extra767
cloudfront_distributions_geo_restrictions_enabled | extra732
cloudfront_distributions_https_enabled | extra738
cloudfront_distributions_logging_enabled | extra714
cloudfront_distributions_using_deprecated_ssl_protocols | extra791
cloudfront_distributions_using_waf | extra773

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: providers/aws/services/cloudfront/cloudfront_distributions_using_deprecated_ssl_protocols/cloudfront_distributions_using_deprecated_ssl_protocols.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from lib.check.models import Check, Check_Report
+from providers.aws.services.cloudfront.cloudfront_client import cloudfront_client
+
+SSLV3 = "SSLv3"
```
Comment (MrCloudSec) [inline providers/aws/services/cloudfront/cloudfront_distributions_using_deprecated_ssl_protocols/cloudfront_distributions_using_deprecated_ssl_protocols.py]: I would put all the ones that accept TLSv1.1 or lower.
More info: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/secure-connections-supported-viewer-protocols-ciphers.html

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
🔝  great job! Please, check my comments.

File: providers/aws/services/cloudfront/cloudfront_distributions_using_deprecated_ssl_protocols/cloudfront_distributions_using_deprecated_ssl_protocols.py (hunk 1 above)
Comment (MrCloudSec) [reply providers/aws/services/cloudfront/cloudfront_distributions_using_deprecated_ssl_protocols/cloudfront_distributions_using_deprecated_ssl_protocols.py]: SSLv3 | TLSv1 | TLSv1_2016 | TLSv1.1_2016
-- | -- | -- | --

Comment (jfagoagas) [reply providers/aws/services/cloudfront/cloudfront_distributions_using_deprecated_ssl_protocols/cloudfront_distributions_using_deprecated_ssl_protocols.py]: Good point! I've updated it, please check it out.

Comment (jfagoagas) [reply providers/aws/services/cloudfront/cloudfront_distributions_using_deprecated_ssl_protocols/cloudfront_distributions_using_deprecated_ssl_protocols.py]: I've changed the logic a little bit

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**